source/slang/slang-state-serialize.cpp

```diff
@@ -1072,10 +1074,234 @@ struct LoadContext
+
+                    String entryPointPath;
+                    if (srcOutputState.outputPath)
+                    {
+                        entryPointPath = base.asRaw(srcOutputState.outputPath)->getSlice();
+                    }
+
+                    dstTargetInfo->entryPointOutputPaths.Add(srcOutputState.entryPointIndex, entryPointPath);
+                }
+            }
+#endif
+        }
+    }
+
+    {
+        const auto& srcPaths = requestState->searchPaths;
+        for (Index i = 0; i < srcPaths.getCount(); ++i)
+        {
+            cmd.addArg("-I");
+            cmd.addArg(base.asRaw(base.asRaw(srcPaths[i]))->getSlice());
+        }
+    }
+
+    {
+        for (const auto& define : requestState->preprocessorDefinitions)
+        {
+            StringBuilder builder;
+            builder << "-D" << base.asRaw(base.asRaw(define).first)->getSlice();
+            if (base.asRaw(define).second)
+            {
+                builder << "=" << base.asRaw(base.asRaw(define).second)->getSlice();
+            }
+        }
+    }
+
+    {
+        const auto& srcTranslationUnits = requestState->translationUnits;
+
+        for (Index i = 0; i < srcTranslationUnits.getCount(); ++i)
+        {
+            const auto& srcTranslationUnit = base.asRaw(srcTranslationUnits[i]);
+
+            //context.loadDefines(srcTranslationUnit.preprocessorDefinitions, dstTranslationUnit->preprocessorDefinitions);
+
+#if 0
+            if (srcTranslationUnit.moduleName)
+            {
+                moduleName = base[srcTranslationUnit].moduleName->getSlice());
+            }
+#endif
+      
+            const auto& srcSourceFiles = srcTranslationUnit.sourceFiles;
+            
+            for (Index j = 0; j < srcSourceFiles.getCount(); ++j)
+            {
+                SourceFileState* sourceFile = base.asRaw(base.asRaw(srcSourceFiles[i]));
+                OffsetString* path = base[sourceFile->foundPath];
+
+                if (path)
+                {
+                    cmd.addArg(path->getSlice());
+                }
+            }
+        }
+    }
+
+    // Entry points
+    {
+        for (const auto& srcEntryPointOffset : requestState->entryPoints)
+        {
+            const auto srcEntryPoint = base.asRaw(srcEntryPointOffset);
+
+            const char* name = srcEntryPoint.name ? base.asRaw(srcEntryPoint.name)->getCstr() : nullptr;
+
+            cmd.addArg("-entry");
+            cmd.addArg(name);
+
+            cmd.addArg("-stage");
+            UnownedStringSlice stageText = getStageText(srcEntryPoint.profile.GetStage());
+            cmd.addArg(stageText);
+
+            //cmd.addArg("-profile");
+            //cmd.addArg(Profile(srcEntryPoint.profile).getName());
+
+
+            //List<const char*> args = context.toList(srcEntryPoint.specializationArgStrings);
+
+            //spAddEntryPointEx(externalRequest, int(srcEntryPoint.translationUnitIndex), name, SlangStage(stage), int(args.getCount()), args.getBuffer());
```

Comment:
It is completely reasonable for this code path to ignore specialization for a good long while (at the very least, until we have an example that requires it). Most interesting cases of specialization are going to need to be driven via API anyway, so eventually we'd hit the limits of what can be expressed by the command-line interface.